File: Src/java/cql-to-elm/src/jvmTest/java/org/cqframework/cql/cql2elm/fhir/v18/PathTests.java
package org.cqframework.cql.cql2elm.fhir.v18;

import static kotlinx.io.CoreKt.buffered;
import static kotlinx.io.JvmCoreKt.asSource;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import java.io.IOException;
import org.cqframework.cql.cql2elm.*;
import org.cqframework.cql.cql2elm.quick.FhirLibrarySourceProvider;
import org.hl7.cql.model.ModelInfoProvider;
import org.hl7.elm.r1.Library;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

/**
 * Created by Bryn on 12/11/2016.
 */
class PathTests {

    private static LibraryManager libraryManager;
    private static ModelManager modelManager;
    private static ModelInfoProvider modelInfoProvider;

    @BeforeAll
    static void setup() {
        modelManager = new ModelManager();
        libraryManager = new LibraryManager(modelManager);
        libraryManager.getLibrarySourceLoader().clearProviders();
        libraryManager.getLibrarySourceLoader().registerProvider(new TestLibrarySourceProvider());
        libraryManager.getLibrarySourceLoader().registerProvider(new FhirLibrarySourceProvider());
        modelInfoProvider = new TestFhirModelInfoProvider(PathTests.class);
        modelManager.getModelInfoLoader().registerModelInfoProvider(modelInfoProvider, true);
    }

    @AfterAll
    static void tearDown() {
        modelManager.getModelInfoLoader().unregisterModelInfoProvider(modelInfoProvider);
    }

    @Test
    void paths() {
        CqlTranslator translator = null;
        try {
            translator = CqlTranslator.fromSource(buffered(asSource(PathTests.class.getResourceAsStream("PathTests.cql"))), libraryManager);
            Library library = translator.toELM();
            assertThat(translator.getErrors().size(), is(0));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Src/java/cql-to-elm/src/jvmTest/kotlin/org/cqframework/cql/cql2elm/fhir/v18/PathTests.kt
package org.cqframework.cql.cql2elm.fhir.v18

import java.io.IOException
import kotlinx.io.asSource
import kotlinx.io.buffered
import org.cqframework.cql.cql2elm.CqlTranslator
import org.cqframework.cql.cql2elm.LibraryManager
import org.cqframework.cql.cql2elm.ModelManager
import org.cqframework.cql.cql2elm.TestFhirModelInfoProvider
import org.cqframework.cql.cql2elm.TestLibrarySourceProvider
import org.cqframework.cql.cql2elm.quick.FhirLibrarySourceProvider
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hl7.cql.model.ModelInfoProvider
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test

@Suppress("PrintStackTrace", "UnusedPrivateProperty", "MaxLineLength")
/** Created by Bryn on 12/11/2016. */
internal class PathTests {
    @Test
    fun paths() {
        var translator: CqlTranslator?
        try {
            translator =
                CqlTranslator.fromSource(
                    PathTests::class
                        .java
                        .getResourceAsStream("PathTests.cql")!!
                        .asSource()
                        .buffered(),
                    libraryManager!!
                )
            val library = translator.toELM()
            assertThat(translator.errors.size, `is`(0))
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    companion object {
        private var libraryManager: LibraryManager? = null
        private var modelManager: ModelManager? = null
        private var modelInfoProvider: ModelInfoProvider? = null

        @JvmStatic
        @BeforeAll
        fun setup() {
            modelManager = ModelManager()
            libraryManager = LibraryManager(modelManager!!)
            libraryManager!!.librarySourceLoader.clearProviders()
            libraryManager!!.librarySourceLoader.registerProvider(TestLibrarySourceProvider())
            libraryManager!!.librarySourceLoader.registerProvider(FhirLibrarySourceProvider())
            modelInfoProvider = TestFhirModelInfoProvider(PathTests::class.java)
            modelManager!!.modelInfoLoader.registerModelInfoProvider(modelInfoProvider!!, true)
        }

        @JvmStatic
        @AfterAll
        fun tearDown() {
            modelManager!!.modelInfoLoader.unregisterModelInfoProvider(modelInfoProvider!!)
        }
    }
}
